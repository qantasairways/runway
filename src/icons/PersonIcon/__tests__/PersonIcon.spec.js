import React from 'react';
import { shallow } from 'enzyme';
import shortid from 'shortid';

import PersonIcon from '..';

jest.mock('shortid');

describe('PersonIcon', () => {
  it('should render', () => {
    shortid.generate.mockReturnValue('1a2b3c');
    const component = shallow(
      <PersonIcon width="36px" height="36px" className="my-favourite-icon" />
    );

    expect(component).toMatchSnapshot();
  });

  it('should set the width', () => {
    const component = shallow(<PersonIcon width="36px" />);
    const result = component.find('svg').props().width;

    expect(result).toEqual('36px');
  });

  it('should set the height', () => {
    const component = shallow(<PersonIcon height="36px" />);
    const result = component.find('svg').props().height;

    expect(result).toEqual('36px');
  });

  it('should set the class name', () => {
    const component = shallow(<PersonIcon className="my-favourite-icon" />);
    const result = component.find('svg').props().className;

    expect(result).toEqual('my-favourite-icon');
  });
});
