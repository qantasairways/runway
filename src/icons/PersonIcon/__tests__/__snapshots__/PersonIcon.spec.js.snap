// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonIcon should render 1`] = `
<svg
  className="my-favourite-icon"
  focusable="false"
  height="36px"
  viewBox="0 0 24 24"
  width="36px"
>
  <defs>
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      id="1a2b3c"
    />
  </defs>
  <g
    fillRule="evenodd"
  >
    <mask
      fill="#fff"
      id="1a2b3c"
    >
      <use
        xlinkHref="#1a2b3c"
      />
    </mask>
    <g
      mask="url(#1a2b3c)"
    >
      <path
        d="M.235.235h23v23h-23z"
      />
    </g>
  </g>
</svg>
`;
